stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/list/adapter/internal/MessageListItemDecoratorProvider.kt

```diff
@@ -51,6 +52,7 @@ internal class MessageListItemDecoratorProvider(
     showAvatarPredicate: MessageListView.ShowAvatarPredicate,
     messageBackgroundFactory: MessageBackgroundFactory,
     deletedMessageListItemPredicate: MessageListView.MessageListItemPredicate,
+    isCurrentUserBanned: () -> Boolean
```

Comment:
We can, but since the check is dependant on the `lateinit var channel: Channel` object that resides inside MessageListView, we can't write a functional interface since we need to pass the channel to it and update it every time live data emits a new value, or we would need to pass the channel object all the way down to the FailedIconDecorator, which doesn't seem like a clean option. I wrote it to reflect `isDirectMessage` check in `MessageListItemDecoratorProvider`. I can refactor it with a regular interface as described before if needed.

Reply:
Just looking at it - yeah it makes sense - isDirectMessage is also here, so we can use this approach!